### ISSUE TITLE
Is there a way to debug the connection?

### ISSUE THREAD

**Issue by RMHogervorst** [sfu-db/connector-x#192]
I've tried connecting to a mssql server like this:

```python
import os
os.environ['RUST_LOG']="connectorx=debug,connectorx_python=debug"
import connectorx as cx

url = "mssql://user_name:password@server_host:port/db_name"
df = cx.read_sql(url, "select top 10 * from dbo.stupidtablename")
```

But all I get back is:
```
[datetime DEBUG connectorx_python::pandas] Running dispatcher
[datetime DEBUG connectorx::dispatcher] Fetching metadata
Traceback to connectorx/__init__py
RuntimeError: Timed out in bb8
```

Is there a way for me to debug connection info?
It must not reach the server or something like that

**Comment by wangxiaoying** [comment]
Hi @RMHogervorst , thanks for the log! Indeed it looks like it cannot connect to mssql. Maybe one thing you can do is to use another client tool to make sure the host and port are correct. What I usually use is [mssql-cli](https://github.com/dbcli/mssql-cli). It is easy to install (`pip install mssql-cli`) and you can try out your connection info using command: `mssql-cli -S"${server_host},${port}" -U${user} -P${password} -d${db_name}`.

**Comment by sundeeprpatil** [comment]
Hi @wangxiaoying 👍  thank you for the advice. 
I dont have issues talking to dB via my laptop (either via pyodbc /  mssql-cli).
I think it's could be some cloud related/ network issue. My code is running as web app service and its quering some data from SQL sitting in a VM. Any advice / comments would help.

**Comment by wangxiaoying** [comment]
Hi @sundeeprpatil , I'm not sure what's the error message you got. But if it's the same timed out, then I think it is because the database is not reachable from the code's running environment. Instead of running on your laptop, you need to run `mssql-cli` inside the same VM to see whether it works.

**Comment by stefVleugels7** [comment]
Hi @sundeeprpatil , I'm experiencing the same issue with an Azure SQL server. Have you found a solution for this?

**Comment by sundeeprpatil** [comment]
Yes: I used IP instead of the right end point, which works locally but does
not work in public cloud.
Check hybrid connection module for the right connection string.
